library-core/src/main/java/com/mikepenz/fastadapter/FastAdapter.kt

```diff
@@ -206,22 +206,31 @@ open class FastAdapter<Item : IItem<out RecyclerView.ViewHolder>> :
      * @param clazz the extension class, to retrieve its instance
      * @return the found IAdapterExtension or null if it is not found
      */
+    @Suppress("UNCHECKED_CAST")
     fun <T : IAdapterExtension<Item>> getExtension(clazz: Class<in T>): T? {
-        return extensionsCache[clazz] as T
+        return extensionsCache[clazz] as T?
     }
 
+
+    inline fun <reified T : IAdapterExtension<Item>> getExtension(): T? = getExtension(T::class.java)
+
+    inline fun <reified T : IAdapterExtension<Item>> requireExtension(): T = getExtension()!!
+
+    @Suppress("UNCHECKED_CAST")
     fun <T : IAdapterExtension<Item>> getOrCreateExtension(clazz: Class<in T>): T? {
         if (extensionsCache.containsKey(clazz)) {
             return extensionsCache[clazz] as T
         }
         val extension = ExtensionsFactories.create(
                 this,
                 clazz as Class<out IAdapterExtension<out IItem<out RecyclerView.ViewHolder>>>
-        ) as? T? ?: return null
+        ) as? T ?: return null
```

Comment:
Removed null check at the end since it's a safe cast anyways. What you expect is a nonnull instance if it conforms